Dockerfile

```diff
@@ -1,38 +1,26 @@
-# Use an official Python base image from the Docker Hub
 FROM python:3.10-slim
 
-# Install git
-RUN apt-get -y update
-RUN apt-get -y install git chromium-driver
+RUN apt-get update
+RUN apt-get install -y git chromium-driver wget gnupg2 libgtk-3-0 libdbus-glib-1-2 dbus-x11 xvfb ca-certificates
 
-# Install Xvfb and other dependencies for headless browser testing
-RUN apt-get update \
-    && apt-get install -y wget gnupg2 libgtk-3-0 libdbus-glib-1-2 dbus-x11 xvfb ca-certificates
-
-# Install Firefox / Chromium
 RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
     && echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google-chrome.list \
     && apt-get update \
     && apt-get install -y chromium firefox-esr
 
-# Set environment variables
 ENV PIP_NO_CACHE_DIR=yes \
     PYTHONUNBUFFERED=1 \
     PYTHONDONTWRITEBYTECODE=1
 
-# Create a non-root user and set permissions
-RUN useradd --create-home appuser
-WORKDIR /home/appuser
-RUN chown appuser:appuser /home/appuser
-USER appuser
+WORKDIR /tmp
+COPY requirements.txt .
+
+# Chop out the part of requirements.txt not intended for the docker image
+RUN sed -i '/Items below this point will not be included in the Docker Image/,$d' requirements.txt
+RUN pip install -r requirements.txt
 
-# Copy the requirements.txt file and install the requirements
-COPY --chown=appuser:appuser requirements.txt .
-RUN sed -i '/Items below this point will not be included in the Docker Image/,$d' requirements.txt && \
-	pip install --no-cache-dir --user -r requirements.txt
+WORKDIR /app
 
-# Copy the application files
-COPY --chown=appuser:appuser autogpt/ ./autogpt
+COPY . .
```

Comment:
We don't need all files in the container build. `docker-compose.yml` contains a non-selective volume mount, that's fine because it is intended for local use. Please be more selective here though, since we will be shipping images using this config.